File: app/src/main/java/com/andreesperanca/deolhonobus/models/BusStop.kt
package com.andreesperanca.deolhonobus.models

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize


@Entity(tableName = "favoritesBusStop")
@Parcelize
data class BusStop(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    @SerializedName("cp")
    val idCodeBusStop: Int,
    @SerializedName("np")
    val name: String,
    @SerializedName("ed")
    val address: String,
    @SerializedName("py")
    val latitude: Double,
    @SerializedName("px")
    val longitude: Double
) : Parcelable

File: lib/models/bus_stop.dart
class BusStop{
  int id;
  String name;
  String address;
  double latitude;
  double longitude;

  BusStop(
      {required this.id,
      required this.name,
      required this.address,
      required this.latitude, 
      required this.longitude});

  factory BusStop.fromJson(Map<String, dynamic> responseData) {
    return BusStop(
        id: responseData['cp'],
        name: responseData['np'],
        address: responseData['ed'],
        longitude: responseData['px'],
        latitude: responseData['py']);
  }
}